runhouse/rns/envs/conda_env.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import subprocess
+from pathlib import Path
+
+from typing import Dict, List, Optional, Union
+
+import yaml
+
+from runhouse.rns.hardware import Cluster
+from runhouse.rns.packages import Package
+
+from .env import _process_reqs, Env
+
+
+class CondaEnv(Env):
+    RESOURCE_TYPE = "env"
+
+    def __init__(
+        self,
+        conda_env: Union[str, Dict],
+        name: Optional[str] = None,
+        reqs: List[Union[str, Package]] = None,
```

Comment:
I currently only have one factory function for creating any env -- `rh.env` checks it theres a `conda_env` arg passed in, and if so, constructs the corresponding `CondaEnv` instead of `Env`. I think it does make sense to additionally support a `rh.conda_env(reqs)`, but naming is vague in this case.

Do you have thoughts on conda env naming -- should the user pass in a `conda_name` for the name of the conda env on the cluster, or we should we create a randomized str name like with our tables, and/or default to the resource name if they give one?